### DISCUSSION TITLE
Use of event modifier

### DISCUSSION THREAD

**Discussion by ariomoklo** [huntabyte/shadcn-svelte#212] [Help] [answered]
I am new to svelte, and found that in svelte there is event modifier that you can chain in some event like `on:click|preventDefault`. But beside `once`, event modifier can be used in DOM elements only. Is there a way to channel the modifiers or some other way to so that event modifier can be used in shadcn components ?

**Comment by edwardspresume** [comment, marked as answer]
> I am new to svelte, and found that in svelte there is event modifier that you can chain in some event like `on:click|preventDefault`. But beside `once`, event modifier can be used in DOM elements only. Is there a way to channel the modifiers or some other way to so that event modifier can be used in shadcn components ?

I am curious if the modifiers can still be used in a simple way as well. For now, using a function is a workaround, for example:

 `function handleClick(event: Event) {
		event.stopPropagation();
		// rest of code
	}`  
	
`<Button
    on:click={handleClick} />`

**Reply by huntabyte** [reply]
I think this is the only way to handle it for component events at the moment!
